### DISCUSSION TITLE
How do you change the path to local bentoml model store?

### DISCUSSION THREAD

**Discussion by rmsiddharth** [bentoml/BentoML#4172] [Q&A] [answered]
The default path for the model store is `/home/user/bentoml` but I would like to change it to a different path. I tried setting `$BENTOML_HOME` to the new path but models, imported to bentoml using the python api, continue to be stored at `/home/user/bentoml`. Is there a way to change the path such that the python api also uses the new path?

Thank you!

**Comment by aarnphm** [comment]
can you share the code that you run with this?

`BENTOML_HOME` should be respected regardless.

**Reply by rmsiddharth** [reply]
I set `BENTOML_HOME` env variable to the new path in my bashrc file. Running any command line instructions appears to use the new path. I imported a model from mlflow using
```
bentoml.mlflow.import_model('model_name', model_uri)
```
which then creates a new directory at the default `/home/user/bentoml` and stores the model under models there. As a result, `bentoml models list` now does not display the model was not imported to `BENTOML_HOME`.

**Reply by aarnphm** [reply]
did you restart your shell environment?

**Reply by rmsiddharth** [reply]
Yes I restarted the shell environment after setting `BENTOML_HOME` to the new path in bashrc and verified that the `BENTOML_HOME` variable is set correctly before executing the rest but the outcome remains the same

**Reply by aarnphm** [reply]
cc @jianshen92 when you have cycle

**Reply by jianshen92** [reply]
Hey, @rmsiddharth I am not able to reproduce your issue. I have set custom path for $BENTOML_HOME and doing mlflow.import_model will indeed import to the new default directory. Could you check in your python session if the environment variable is being set? 

<img width="1268" alt="Screenshot 2023-09-15 at 8 10 59 PM" src="https://github.com/bentoml/BentoML/assets/28883559/6591ad4a-d87f-4195-a35a-a6498602e4d4">

**Reply by rmsiddharth** [reply, marked as answer]
Hey @jianshen92! It appears to be working as expected. Like you said, it appears that my jupyter notebook in vscode does not see all the environment variables. I tried it in a separate script and the $BENTOML_HOME variable is respected as expected. Thanks for the help! :)
